src/fallback.ts

```diff
@@ -144,12 +146,34 @@ export class GrpcClient {
   ) {
     function buildMetadata(abTests, moreHeaders) {
       const metadata = {};
+      if (!headers) {
+        headers = {};
+      }
+      for (const key in headers) {
```

Comment:
perhaps a comment here, is this to support headers in both Array and Objet form?

Reply:
added comment.